simulator/risc_v/riscv_decoder.h

```diff
@@ -106,25 +228,36 @@ struct RISCVInstrDecoder
-        , B_imm11   ( apply_mask( raw, 0b00000000'00000000'00000000'10000000))
-        , B_imm4_1  ( apply_mask( raw, 0b00000000'00000000'00001111'00000000))
-        , B_imm10_5 ( apply_mask( raw, 0b01111110'00000000'00000000'00000000))
-        , B_imm12   ( apply_mask( raw, 0b10000000'00000000'00000000'00000000))
-        , U_imm     ( apply_mask( raw, 0b11111111'11111111'11110000'00000000))
-        , J_imm19_12( apply_mask( raw, 0b00000000'00001111'11110000'00000000))
-        , J_imm11   ( apply_mask( raw, 0b00000000'00010000'00000000'00000000))
-        , J_imm10_1 ( apply_mask( raw, 0b01111111'11100000'00000000'00000000))
-        , J_imm20   ( apply_mask( raw, 0b10000000'00000000'00000000'00000000))
-        , csr_imm   ( apply_mask( raw, 0b00000000'00001111'10000000'00000000))
-        , csr       ( apply_mask( raw, 0b11111111'11110000'00000000'00000000))
-        , bytes     ( apply_mask( raw, 0b11111111'11111111'11111111'11111111))
+        : sz             ( apply_mask( raw, 0b00000000'00000000'00000000'00000011))
+        , rd             ( apply_mask( raw, 0b00000000'00000000'00001111'10000000))
+        , rs1            ( apply_mask( raw, 0b00000000'00001111'10000000'00000000))
+        , rs2            ( apply_mask( raw, 0b00000001'11110000'00000000'00000000))
+        , rs2_compr      ( apply_mask( raw, 0b00000000'00000000'00000000'01111100))
+        , rd_3_bits      ( apply_mask( raw, 0b00000000'00000000'00000000'00011100))
+        , rs1_3_bits     ( apply_mask( raw, 0b00000000'00000000'00000011'10000000))
+        , rs2_3_bits     ( apply_mask( raw, 0b00000000'00000000'00000000'00011100))
+        , I_imm          ( apply_mask( raw, 0b11111111'11110000'00000000'00000000))
+        , S_imm4_0       ( apply_mask( raw, 0b00000000'00000000'00001111'10000000))
+        , S_imm11_5      ( apply_mask( raw, 0b11111110'00000000'00000000'00000000))
+        , B_imm11        ( apply_mask( raw, 0b00000000'00000000'00000000'10000000))
+        , B_imm4_1       ( apply_mask( raw, 0b00000000'00000000'00001111'00000000))
+        , B_imm10_5      ( apply_mask( raw, 0b01111110'00000000'00000000'00000000))
+        , B_imm12        ( apply_mask( raw, 0b10000000'00000000'00000000'00000000))
+        , U_imm          ( apply_mask( raw, 0b11111111'11111111'11110000'00000000))
+        , J_imm19_12     ( apply_mask( raw, 0b00000000'00001111'11110000'00000000))
+        , J_imm11        ( apply_mask( raw, 0b00000000'00010000'00000000'00000000))
+        , J_imm10_1      ( apply_mask( raw, 0b01111111'11100000'00000000'00000000))
+        , J_imm20        ( apply_mask( raw, 0b10000000'00000000'00000000'00000000))
+        , csr_imm        ( apply_mask( raw, 0b00000000'00001111'10000000'00000000))
+        , csr            ( apply_mask( raw, 0b11111111'11110000'00000000'00000000))
+        , bytes          ( apply_mask( raw, 0b11111111'11111111'11111111'11111111))
+        , Compr_imm12    ( apply_mask( raw, 0b00000000'00000000'00010000'00000000))
```

Comment:
I suggest to trim higher 16 bits to highlight that masks are related to 16-bit instructions.